include/brickengine/collision_return_values.hpp

```diff
@@ -0,0 +1,11 @@
+struct collisionReturnValues{
+    double spaceLeft;
+    int objectId;
+    collisionReturnValues(double spaceLeft, int objectId) : spaceLeft(spaceLeft), objectId(objectId){}
+};
+
+struct triggerReturnValues{
+    bool isInTrigger;
+    int objectId;
```

Comment:
I would suggest making this integer optional, to make the intent more clear that can be no object colliding right now.